lib/shelf_io.dart

```diff
@@ -31,6 +31,52 @@ import 'src/util.dart';
 
 export 'src/io_server.dart' show IOServer;
 
+/// Starts an [HttpServer] that listens on the specified [address] and
+/// [port] and sends requests to [handler].
+///
+/// If a [securityContext] is provided an HTTPS server will be started.
+///
+/// Serves requests in an error zone, using [onError] to handle errors, if
+/// provided. Uses a default error handler that prints error summary if
+/// [onError] is not provided.
+///
+/// See the documentation for [HttpServer.bind] and [HttpServer.bindSecure]
+/// for more details on [address], [port], [backlog], and [shared].
+Future<HttpServer> serveGuarded(
+  Handler handler,
+  address,
+  int port, {
+  SecurityContext? securityContext,
+  int? backlog,
+  bool shared = false,
+  void Function(Object, StackTrace) onError = _logServeRequestErrors,
+}) async {
+  backlog ??= 0;
+  var server = await (securityContext == null
```

Comment:
This can also fail - it should be ran in an error handling zone as well

Reply:
This error would surface in the returned `Future` instead of disappear as uncaught async, so I think it's OK to let it skip the `onError` callback. Although I also think we should consider dropping the method entirely and push the `HttpServer.bind` call into call-site - but we should look at more usage to figure out if that is sensible.